src/AEG/AEGTree.ts

```diff
@@ -50,4 +68,36 @@ export class AEGTree {
     public remove(incomingPoint: Point): void {
         this.sheet.remove(incomingPoint);
     }
+
+    /**
+     * Determines if the incoming node's boundaries overlap the other node's boundaries.
+     * @param incomingNode the incoming node
+     * @param otherNode the other node
+     * @returns the result of each shape's respective overlaps() methods.
+     */
+    public overlaps(incomingNode: AtomNode | CutNode, otherNode: AtomNode | CutNode): boolean {
```

Comment:
This should probably be a private helper method, I can't see an external use case for checking if your own internal nodes are overlapping.